src/compiler/checker.ts

```diff
@@ -11890,6 +11890,11 @@ namespace ts {
                     // If the effective argument type is 'undefined', there is no synthetic type
                     // for the argument. In that case, we should check the argument.
                     if (argType === undefined) {
+                        // If the parameter and argument are both functions and the parameter has fewer arguments than the argument,
+                        // then this signature is not applicable. Exit early.

```

Comment:
"Exit early to avoid fixing incorrect contextual types to the function expression parameters"